packages/cms-data-sync/src/users/users.data-migration.ts

```diff
@@ -73,15 +81,20 @@ export const migrateUsers = async () => {
     if (avatar) {
       const urls = createAssetUrl(avatar.map((asset) => asset.id));
       if (urls.length > 0) {
-        avatarAsset = await createAsset(contentfulEnvironment, [
-          { ...avatar[0], thumbnailUrl: urls[0] },
-        ]);
+        avatarAsset = await createAsset(
+          contentfulEnvironment,
+          [{ ...avatar[0], thumbnailUrl: urls[0] }],
+          limiter,
+        );
       }
     }
 
     const teamLinks = await Promise.all(
       (teams || []).map(async (team) => {
         if (team.id && team.id.length) {
+          // create and publish
+          await limiter.removeTokens(2);
```

Comment:
i think we could easily hit the rate limit here due to the use of `map` as it will execute at the same time. I think using a `for-of` loop will solve the potential issue

Reply:
actually i think that would be the case before the use of the rate limiter, but with that tool now it should control the flow, although all of the async functions will be run at the same time (sort of) each of them will require awaiting of the removal of tokens which means that if we hit the limit that promise will not resolve until there are tokens available again